pkg/util/winutil/servicemain/servicemain.go

```diff
@@ -23,6 +25,32 @@ import (
 	"golang.org/x/sys/windows/svc"
 )
 
+const (
+	// DefaultHardStopTimeout is a default value. See Service.HardStopTimeout for details.
+	DefaultHardStopTimeout = 15 * time.Second
+	// EnvHardStopTimeoutOverride is an environment variable that a user can set
+	// to override DefaultHardStopTimeout.
+	EnvHardStopTimeoutOverride = "DD_WINDOWS_SERVICE_STOP_TIMEOUT_SECONDS"
+)
+
+// DefaultSettings provides default values to Service implementations when embedded
+type DefaultSettings struct{}
+
+// HardStopTimeout provides a default hard stop timeout for Service implementations,
+// and allows a user to override the default by setting the DD_WINDOWS_SERVICE_STOP_TIMEOUT_SECONDS
+// environment variable.
+func (s *DefaultSettings) HardStopTimeout() time.Duration {
+	timeString, found := os.LookupEnv(EnvHardStopTimeoutOverride)
```

Comment:
Any reason for not using the `config` package instead as we do for the rest of env variables in the Agent?

Declaring a `windows_service_stop_timeout_seconds` configuration flag would provide you the `DD_WINDOWS_SERVICE_STOP_TIMEOUT_SECONDS` environment variable for free. It would also let you configure a default value.

See: https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config.go#L1252 (took a random example)

Reply:
This code must run before the agent components come up, so I didn't think I would have access to the config. I see that it's still also a global that's initialized in `init`, do we expect that to continue to be the case or do we intend to remove the Config globals in favor of components?
https://github.com/DataDog/datadog-agent/blob/4e241a7e36f6b4aea877aefa2fb4cae2f1c7e5f5/pkg/config/config.go#L102-L106

Also, the servicemain package is shared between the different agents/services in this repo, and in some WKIT repos, so it might not make sense to reference the core agent config object here.

Reply:
> This code must run before the agent components come up,

I see 👍 

> do we expect that to continue to be the case or do we intend to remove the Config globals in favor of components?

I expect us to get rid of the globals at some point but I don't have the actual answer, Agent-Shared-Components will know better. I send the ✅ (I don't want to block for 7.50.0!) but please think about making sure that there is no better alternative with the config pkg, having environment variables scattered here and there in the sources isn't a good thing.

Reply:
opened a card to track it https://datadoghq.atlassian.net/browse/WINA-524